core/src/main/scala/cats/data/NonEmptyChain.scala

```diff
@@ -0,0 +1,456 @@
+
+      def flatMap[A, B](fa: NonEmptyChain[A])(f: A => NonEmptyChain[B]): NonEmptyChain[B] =
+        fa.flatMap(f)
+
+      def tailRecM[A, B](a: A)(f: A => NonEmptyChain[Either[A, B]]): NonEmptyChain[B] =
+        create(Monad[Chain].tailRecM(a)(a => unwrap(f(a))))
+
+      def extract[A](x: NonEmptyChain[A]): A = x.head
+
+      def coflatMap[A, B](fa: NonEmptyChain[A])(f: NonEmptyChain[A] => B): NonEmptyChain[B] = {
+        @tailrec def go(as: Chain[A], res: ListBuffer[B]): Chain[B] =
+          as.uncons match {
+            case Some((h, t)) => go(t, res += f(NonEmptyChain.fromChainPrepend(h, t)))
+            case None => Chain.fromSeq(res.result())
+          }
+        NonEmptyChain.fromChainPrepend(f(fa), go(fa.tail, ListBuffer.empty))
+      }
+
+      def nonEmptyTraverse[G[_]: Apply, A, B](fa: NonEmptyChain[A])(f: A => G[B]): G[NonEmptyChain[B]] =
+        Foldable[Chain].reduceRightToOption[A, G[Chain[B]]](fa.tail)(a => Apply[G].map(f(a))(Chain.one)) { (a, lglb) =>
+          Apply[G].map2Eval(f(a), lglb)(_ +: _)
+        }.map {
+          case None => Apply[G].map(f(fa.head))(NonEmptyChain.one)
+          case Some(gtail) => Apply[G].map2(f(fa.head), gtail)((h, t) => create(Chain.one(h) ++ t))
+        }.value
+
+      override def size[A](fa: NonEmptyChain[A]): Long = fa.length
+
+      override def reduceLeft[A](fa: NonEmptyChain[A])(f: (A, A) => A): A =
+        fa.reduceLeft(f)
+
+      override def reduce[A](fa: NonEmptyChain[A])(implicit A: Semigroup[A]): A =
+        fa.reduce
+
+      def reduceLeftTo[A, B](fa: NonEmptyChain[A])(f: A => B)(g: (B, A) => B): B = fa.reduceLeftTo(f)(g)
+
+      def reduceRightTo[A, B](fa: NonEmptyChain[A])(f: A => B)(g: (A, Eval[B]) => Eval[B]): Eval[B] =
+        Eval.defer(fa.reduceRightTo(a => Eval.now(f(a))) { (a, b) =>
+          Eval.defer(g(a, b))
+        })
+
+      override def foldLeft[A, B](fa: NonEmptyChain[A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b)(f)
+
+      override def foldRight[A, B](fa: NonEmptyChain[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+        fa.foldRight(lb)(f)
+
+      override def foldMap[A, B](fa: NonEmptyChain[A])(f: A => B)(implicit B: Monoid[B]): B =
+        B.combineAll(fa.toChain.iterator.map(f))
+
+      override def fold[A](fa: NonEmptyChain[A])(implicit A: Monoid[A]): A =
+        fa.reduce
+
+      override def find[A](fa: NonEmptyChain[A])(f: A => Boolean): Option[A] =
+        fa.find(f)
+
+      override def forall[A](fa: NonEmptyChain[A])(p: A => Boolean): Boolean =
+        fa.forall(p)
+
+      override def exists[A](fa: NonEmptyChain[A])(p: A => Boolean): Boolean =
+        fa.exists(p)
+
+      override def toList[A](fa: NonEmptyChain[A]): List[A] = fa.toChain.toList
+
+      override def toNonEmptyList[A](fa: NonEmptyChain[A]): NonEmptyList[A] =
+        fa.toNonEmptyList
+    }
+
+  implicit def catsDataEqForNonEmptyChain[A: Eq]: Eq[NonEmptyChain[A]] =
```

Comment:
we only have `Eq` here, no Order, PartialOrder, right? Can we add those so we get the priority right there?